jolt-core/src/test/resources/json/templatr/functions/arrayTests.json

```diff
@@ -0,0 +1,46 @@
+{
+    "input": [
+        {
+            "value": 2
+        },
+        {
+            "value": 2.0
+        }
+    ],
+
+    "spec": {
+        "*": {
+            "max": "=maxOf(@(2,[&1].value),^value,0,1l)",
```

Comment:
1l means a 1 that is a Java Long.   There is no good way for that to be specified via Json, so all the Long, Double, Int stuff should be unit tested at the Math.max Function level, where you have TestNG Java DataProvider that can make the Longs, Ints, and Double.